### PR TITLE
New titles

### PR DESCRIPTION
**Pull request by Shiranui-Izayoi**
Added a total of 16 new titles to the game.
9 universal titles that are always available those being `Switch`, `Kitten`, `Puppy`, `Foxy`, `Bunny`, `Doll`, `Demon`, `Angel` & `Succubus`
1 stable/pony title that being `Majestic Alicorn`
4 ABDL titles those being `Little One`, `Baby`, `Diaper Lover` & `Bondage Baby`
1 title for beating Sophie that being `Duchess` title is not given retroactively & adds a new log to if you win the kidnap fight with her.
1 new maid title that being `Bondage Maid`

Modified a few of the Pony titles. I did this in order to fit the new one in. `Unicorn` is now level 7 & `Pegesus` is 8 & 9 with `Alicorn` being 10. Also modified the `Foal` title so that it can now be used if you are either `Dressage` 1 as it was before but also if you have `ABDL` rep & are at least `Dressage` 1. I did this since I have seen a few people on club that would use it due to being both diapered & in pony gear.

The requirements for the ABDL titles are all 1 or more `ABDL` rep since it is currently very hard & expensive to gain ABDL rep. The only methods being getting diapered and giving a cute response that will cost 10 & 5 money each time for 1 rep. The `Bondage Baby` title also require `Evasion` 10 to be earned.

The `Maid` title requires that you have either joined or are leading the maids & have an `Evasion` of 10.

Also added the `Earned` attribute to all titles that aren't always able to be used or are forced onto the player. This causes the title to take on the same colour as the players label colour. After testing even solid white aka #ffffff is still quite readable.

### CODE REVIEW

**Comment by Ben987** [comment]
@GIT4NICK , this change reworks some of the pony titles.  Can you validate it and let us know if you're ok with it?

@Shiranui-Izayoi , why would you need an evasion skill at 10 to have the maid title?  Thanks.

**Comment by Shiranui-Izayoi** [comment]
> 
> 
> @GIT4NICK , this change reworks some of the pony titles. Can you validate it and let us know if you're ok with it?
> 
> @Shiranui-Izayoi , why would you need an evasion skill at 10 to have the maid title? Thanks.

The reason I required the evasion skill of 10 for the maid title is that it is meant to mean that the player is known for both being tied up a lot and also a maid. So I felt that if someone is tied up a lot that they should also be really good at getting free.

I also didn't feel that Bondage or negative dom rep really fit. Does that answer your question? If you feel like it I could lower it.

**Comment by Ben987** [comment]
I don't understand why you would need the evasion skill to be a maid.  What about head maid, do you also need evasion or another skill?

**Comment by Shiranui-Izayoi** [comment]
> 
> 
> I don't understand why you would need the evasion skill to be a maid. What about head maid, do you also need evasion or another skill?

No no you misunderstand. It isn't a replacement title it is a new title that you can earn in addition to Maid & Head Maid.

Bondage Maid is meant to be a secondary title you can earn if you have both high evasion AND have either joined the Maids or are Head Maid. If you are a Head Maid and have evasion 10 you could either use the title Head Maid or the title Bondage Maid title. As both can be had at once.

Basically the point of this pull request was to add more titles to the game for people to earn. I hope that makes sense.

**Comment by Ben987** [comment]
I see!  I got confused by this phrase: "The Maid title requires that you have either joined or are leading the maids & have an Evasion of 10."  It's not the "maid" but the "bondage maid".  Alright, let's wait for git4nick for the stable titles.

**Comment by Nina-1474** [comment]
3 months passed with no response from git4nick on #1037, so I don't think he's seeing these any more.

**Comment by Ben987** [comment]
@GIT4NICK , can you review and approve these changes please?  

If he doesn't answer from here, I'll try to reach him on DeviantArt.

**Comment by Shiranui-Izayoi** [comment]
@Ben987 Since it has been 10 days and it doesn't look like GIT4NICK has even seen this could you try to reach them on DeviantArt or if not could we start on the review stage?

**Comment by Ben987** [comment]
I could not reach git4nick but I'm pretty sure he would be fine with it.  Can you resolve the conflict from the other pull requests and test it on your side to see if everything still works?  Let me know when it's ready to merge, thanks!

**Comment by Shiranui-Izayoi** [comment]
> 
> 
> I could not reach git4nick but I'm pretty sure he would be fine with it. Can you resolve the conflict from the other pull requests and test it on your side to see if everything still works? Let me know when it's ready to merge, thanks!

Ok fixed the conflict the only thing I needed to do is change DrawText to DrawTextFit really. I did a quick test and it seems everything is working fine. Did catch one accidental super minor thing I forgot not really a bug so much as I forgot to give the forced titles the grey backdrop thing.

It should be ready to go, as everything seems to be working.

**Comment by Ben987** [comment]
Awesome!  I will test it soon.  Thanks.
